src/term/mod.rs

```diff
@@ -96,6 +98,48 @@ impl Search for Term {
         point
     }
 
+    fn bracket_pair_search(&self, start: Point<usize>, direction: Side, start_char: char, needle: char) -> Option<Point<usize>> {
```

Comment:
I'm not entirely happy with the verbosity of this method, so if anyone knows of a better implementation (potentially using a more functional style), let me know!